src/main/java/pl/kinga/produkty/ProductController.java

```diff
@@ -0,0 +1,49 @@
+package pl.kinga.produkty;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+
+@Controller
+public class ProductController {
+    private ProductRepository productRepository;
+    List<Product> products;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model) {
+        return "home";
+    }
+
+    @GetMapping("/allProducts")
+    public String product(Model model) {
+        products = productRepository.findAll();
+        model.addAttribute("product", new Product());
+        model.addAttribute("products", products);
+        return "AllProducts";
+    }
+
+//    @GetMapping("/category")
+//    public String productByCategory(@RequestParam String category, Model model) {
```

Comment:
Wyświetlanie wszzystkich prodduktów jak i produktów w wybranej kategorii mozesz obsluzyc w jednej metodzie, przyjmuj tutaj parametr kategorrii - niech bedzie niewymagany. Jesli parametr jest podany to pobierz z repozytorium produkty z danej kategorii i dodaj je do modelu, w przeciwnym razie pobierz wszystkie produkty i dodaj je do modelu. Pamiętaj jeszcze o zsumowaniu cen potem ;)